web/views/api/helpers.ex

```diff
@@ -0,0 +1,52 @@
+defmodule Castle.API.Helpers do
+
+  def meta_json(json, %{interval: interval, meta: meta}) do
+    json |> Map.merge(%{interval: interval, meta: meta})
+  end
+  def meta_json(json, %{meta: meta}) do
+    json |> Map.put(:meta, meta)
+  end
+
+  def counts_json(json, %{downloads: downloads}) do
+    json |> Map.put(:downloads, Enum.map(downloads, &count_json/1))
+  end
+  def counts_json(json, %{impressions: impressions}) do
+    json |> Map.put(:impressions, Enum.map(impressions, &count_json/1))
+  end
+
+  defp count_json(data), do: [data.time, data.count]
+
+  def groups_json(json, %{downloads: downloads}) do
+    json |> group_names(downloads) |> Map.put(:downloads, grouped(downloads))
+  end
+  def groups_json(json, %{impressions: impressions}) do
+    json |> group_names(impressions) |> Map.put(:impressions, grouped(impressions))
+  end
+
+  defp group_names(json, data) do
+    names = group_ranks(data) |> Enum.map(&group_find_rank(data, &1))
+    json |> Map.put(:groups, names)
+  end
+
+  defp group_ranks(data) do
+    data |> Enum.map(&(&1.rank)) |> Enum.uniq() |> Enum.sort()
+  end
+
+  defp group_find_rank(data, rank) do
+    Enum.find_value data, "Other", &(if &1.rank == rank do &1.display end)
+  end
+
+  defp grouped(raw_data) do
+    ranks = group_ranks(raw_data)
+    raw_data
```

Comment:
This was actually a little painful.  Trying to turn BigQuery results like:

```elixir
[
  %{count: 123, display: nil, rank: nil, time: "2017-07-15T00:00:00Z"},
  %{count: 4271, display: "United States", rank: 1, time: "2017-07-15T00:00:00Z"},
  %{count: 85, display: "United Kingdom", rank: 2, time: "2017-07-15T00:00:00Z"},
  %{count: 169, display: nil, rank: nil, time: "2017-07-15T01:00:00Z"},
  %{count: 4335, display: "United States", rank: 1, time: "2017-07-15T01:00:00Z"},
  %{count: 71, display: "United Kingdom", rank: 2, time: "2017-07-15T01:00:00Z"},
]
```

Into a flat array, grouped by time:

```elixir
[
  ["2017-07-15T00:00:00Z", 4271, 85, 123],
  ["2017-07-15T01:00:00Z", 4335, 71, 169],
]
```

So that they match up with the groups also returned by the API:

```elixir
groups: ["United States", "United Kingdom", "Other"]
```